container/container.go

```diff
@@ -86,19 +92,26 @@ func StopContainer(client DockerClient, containerId string, timeout uint) error
 // credentials passed in.
 func PullImage(client DockerClient, taskInfo *mesos.TaskInfo, authConfig *docker.AuthConfiguration) error {
 	log.Infof("Pulling Docker image '%s'", taskInfo.Container.Docker.Image)
-	err := client.PullImage(
-		docker.PullImageOptions{
-			Repository: taskInfo.Container.Docker.Image,
-		},
-		*authConfig,
-	)
-	if err != nil {
-		return err
-	}
 
-	log.Info("Pulled.")
+	err := retry.Do(func() error {
+		err := client.PullImage(
+			docker.PullImageOptions{
+				Repository: taskInfo.Container.Docker.Image,
+			},
+			*authConfig,
+		)
+		if err != nil {
+			return err
```

Comment:
Does this log fetch errors somehow? Otherwise, it might be good to log how many times it had to retry before it managed to fetch it.

Reply:
I don't think it bubbles up anywhere - good point. I'll add a log entry.

Reply:
Updated.

Reply:
Thanks, looks great! :)